src/run/ci_provider/buildkite/provider.rs

```diff
@@ -127,10 +128,14 @@ impl CIProvider for BuildkiteProvider {
         "Buildkite"
     }
 
-    fn get_provider_slug(&self) -> &'static str {
+    fn get_platform(&self) -> &'static str {
         "buildkite"
     }
 
+    fn get_platform_slug(&self) -> PlatformSlug {
+        PlatformSlug::Buildkite
+    }
+
```

Comment:
This seems not correct. The meaning of slug here is a stringified unique identifier. 
So having a rust enum named PlatformSlug seem to be an antipattern. If serde does the slugification then, this should not be suffixed with slug.

Reply:
Also, what's the motivation behind renaming the provider into the platform ? And having both something to get the slug and something to get the enum(this seems redundant)?

Reply:
As seen together, we will keep `platform` in the runner, but serialize it to `platformSlug` in order to match the changes in our DB.